packages/fast-components-react-base/src/checkbox/checkbox.tsx

```diff
@@ -87,6 +98,36 @@ class Checkbox extends Foundation<ICheckboxHandledProps & ICheckboxManagedClasse
 
         return super.generateClassNames(classes);
     }
+
+    /**
+     * Stores HTML tag for use in render
+     */
+    private get tag(): string {
+        return CheckboxHTMLTags[this.props.tag] || CheckboxHTMLTags.label;
+    }
+
+    /**
+     * Apply indeterminate state to items that are indeterminate.
+     * This method should be called after render because it relies on element references.
+     */
+    private applyIndeterminateState(): void {
+         if (!isUndefined(this.props.indeterminate) && this.inputRef.current && this.state.checked) {
+            this.inputRef.current.indeterminate = this.props.indeterminate;
+        }
+    }
+
+    /**
+     * Handles onChange as a controlled component
+     */
+    private handleCheckboxChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
+        if (isUndefined(this.props.checked)) {
+            this.setState({checked: !this.state.checked});
+        }
+
+        if (this.props.onChange) {
+            this.props.onChange(e);
+        }
```

Comment:
there should be a check when props are updated to also change the state since checked is based only on state, you can use this lifecycle hook: https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops